File: windows/SSH.Test/Logger.cs
using SSH.Core;
using System;

namespace SSH.Test
{
    public class Logger : ILogger
    {
        public void Log(string level, string message)
        {
            var output = $"[{level}] {message}";
            System.Diagnostics.Debug.WriteLine(output);
            Console.WriteLine(output);
        }

        public void Debug(string message)
        {
            Log("Debug", message);
        }

        public void Info(string message)
        {
            Log("Info", message);
        }

        public void Warn(string message)
        {
            Log("Warn", message);
        }

        public void Error(string message)
        {
            Log("Error", message);
        }

        public void Fatal(string message)
        {
            Log("Fatal", message);
        }
    }
}

File: android/app/src/main/java/com/db_explorer/modules/Logger.kt
package com.db_explorer.modules

class Logger {
    fun log(level: String, message: String) {
        // Send to React Native
        PlatformDriverManager.Current?.emit(
          DriverManagerOutboundMessage(
              DriverManagerOutboundMessageType.Log,
              mapOf("level" to level, "message" to message),
          ),
        )
    }

    fun debug(message: String) {
        log("Debug", message)
    }

    fun info(message: String) {
        log("Info", message)
    }

    fun warn(message: String) {
        log("Warn", message)
    }

    fun error(message: String) {
        log("Error", message)
    }

    fun fatal(message: String) {
        log("Fatal", message)
    }
}